filebeat/input/log/log.go

```diff
@@ -111,16 +116,22 @@ func (f *Log) errorChecks(err error) error {
 		return err
 	}
 
+	// At this point we have hit EOF!
+
 	// Stdin is not continuable
 	if !f.fs.Continuable() {
 		logp.Debug("harvester", "Source is not continuable: %s", f.fs.Name())
 		return err
 	}
 
-	if err == io.EOF && f.config.CloseEOF {
+	if f.config.CloseEOF {
 		return err
 	}
 
+	return nil
+}
+
+func (f *Log) checkFileErrors() error {
```

Comment:
Unfortunately github doesn't allow me to comment in code.
With this change every single read is preceded by an extra syscall, to check the file state. Maybe we can reduce the check to be only run if CloseRemoved or CloseRenamed is set.

The CloseInactive check MUST NOT be executed here. This function is run before the next line is read. If the harvester was blocked by the outputs for some amount of time, we might trigger an `ErrInactive` here, although the file was still written to. The CloseInactive check is supposed to trigger only after we did reach EOF.

The truncate case is somewhat tricky. I think checking truncate after EOF should be ok, as you should get a return of 0bytes read + EOF if you read on a truncated file. The check (and any synchronous check alike) is always subject to false-negatives, if the output was blocked for such a long time that the file was allowed to grow past offset.

By moving truncate and CloseInactive checks back to `errorCheck`, we don't need to execute the stat syscall for everyone line we want to read if CloseRemoved or CloseRenamed are not set.